### PR TITLE
[feat] Consolidate SmartRate functions, add new functions

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add new `SmartRate` service for interacting with the SmartRate API
  - New `EstimateDeliveryDate` function to estimate delivery date based on a list of carriers, to/from ZIP codes and a planned ship date (no existing shipment required).
  - New `RecommendShipDate` function to to recommend ship date based on a list of carriers, to/from ZIP codes and a planned ship date (no existing shipment required).
- New `EstimateDeliveryDate` function in `Shipment` service to estimate delivery date for a shipment based on a planned ship date.
- New `RecommendShipDateForShipment` function in `Shipment` service to recommend ship date for a shipment based on a desired delivery date.
 - New model classes as needed for JSON response to new API functions

# Testing

- Unit tests added, updated accordingly
- Cassettes added, re-recorded accordingly

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by Justintime50 — COMMENTED** [review]
I think this looks good. As discussed, we'll hold off until this goes GA to merge in and release. I feel splitting SmartRate to its own service is a good idea especially as the product matures and more functionality becomes available. Since everything is static now, it doesn't make sense to keep it in the Shipment service because you have to pass in the shipment ID anyway now, it no longer inherently belongs to Shipment.

File: EasyPost/Services/Beta/SmartRateService.cs
Comment (jchen293) [inline EasyPost/Services/Beta/SmartRateService.cs:16]: Should this class be out of beta when the endpoints are in GA?

File: EasyPost/Services/Beta/SmartRateService.cs
Comment (jchen293) [inline EasyPost/Services/Beta/SmartRateService.cs:44]: URL seems incorrect to what this function does, you might want to add a TODO and come back to fix it when API docs is added for this endpoint

**Review by jchen293 — COMMENTED** [review]
Code seems good, just two questions. I will approve when the endpoints are in GA

**Comment by Justintime50** [comment]
I'd like this PR to follow the conventions set in the Python one so the experience is consistent across libs, namely the location of the functions in two separate services (keeping the existing functions in the Shipment service), renaming them, and not overloading the functions even though that's a feature of this language (to ensure consistency).

File: EasyPost.Tests/ServicesTests/WithParameters/ShipmentServiceTest.cs
Comment (jchen293) [inline EasyPost.Tests/ServicesTests/WithParameters/ShipmentServiceTest.cs:352]: This unit test needs to be removed as we went with the Python lib approach

File: CHANGELOG.md
Comment (jchen293) [inline CHANGELOG.md:11]: This needs to be updated as we went with the Python lib approach

File: EasyPost/Models/API/EstimateDeliveryDateForZipPairResult.cs
Comment (jchen293) [inline EasyPost/Models/API/EstimateDeliveryDateForZipPairResult.cs:48]: Our API response is `Results`, we should not change it to `Estimates` in C# exclusively, we should be consistent with other libs. This also apply to other fields (from_zip, to_zip, and carriers_without_tint_estimates)

File: EasyPost/Models/API/TimeInTransitDetailsForShipDate.cs
Comment (jchen293) [inline EasyPost/Models/API/TimeInTransitDetailsForShipDate.cs:29]:
```suggestion
    [JsonProperty("days_in_transit")]
    public TimeInTransit? DayInTransit { get; set; }
```

**Review by jchen293 — CHANGES_REQUESTED** [review]

File: EasyPost/BetaClient.cs
Comment (nwithan8) [reply EasyPost/BetaClient.cs:43]: We are not going to deprecate or remove those beta functions in this current PR, that's a separate task.

File: EasyPost/BetaClient.cs
Comment (nwithan8) [reply EasyPost/BetaClient.cs:33]: No, it's just to clean up the import by making the import path more explicit.

File: CHANGELOG.md
Comment (jchen293) [inline CHANGELOG.md:11]: Please update the changelog, the function name doesn't match what we have and we are not deprecating the existing function in `Shipment` service. I will approve once this is addressed

**Review by jchen293 — APPROVED** [review]

**Pull request merged**
